File: 06 - DAT203.1x - Data Science Essentials/Labs/Mod5/PlotCols.py
def auto_scatter_simple(df, plot_cols):
    import matplotlib.pyplot as plt
    ## Loop over the columns
    for col in plot_cols:
        fig = plt.figure(figsize=(6, 6))
        ax = fig.gca()
        ## simple scatter plot
        df.plot(kind = 'scatter', x = col, y = 'lnprice', 
                           ax = ax, color = 'DarkBlue')                          
        ax.set_title('Scatter plot of lnprice vs. ' + col)
        fig.savefig('scatter_' + col + '.png')
    return plot_cols              

def azureml_main(df): 
    import matplotlib
    matplotlib.use('agg')
    ## Define plot columns
    plot_cols = ["weight",
               "enginesize",
               "citympg"]
    auto_scatter_simple(df, plot_cols) ## Create plots
    return df

File: 06 - DAT203.1x - Data Science Essentials/Labs/Mod5/PlotCols.R
vis.simple <- function(col1, df = auto.price, col2 = 'lnprice'){
  require(ggplot2) # make sure ggplot2 is loaded
  title = paste('Plot of', col1, 'vs.', col2) # create the title text string
  ggplot(df, aes_string(col1, col2)) +
    geom_point() +
    ggtitle(title)
}

## Scatter plots of select columns
auto.price <- maml.mapInputPort(1) # read autos data frame from port 1
plot.cols <- c("weight",
               "enginesize",
               "citympg")
lapply(plot.cols, vis.simple) # apply the plot columns to the function
maml.mapOutputPort("auto.price") ## Output the data frame


